### ISSUE TITLE
Unclear error message when AWS credentials are missing in validate

### ISSUE THREAD

**Issue by ankon** [coreos/kube-aws#202]
Error from 0.9.3-rc.2:
~~~~
$ kube-aws validate
Validating UserData...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x495ed8]

goroutine 1 [running]:
panic(0xa16260, 0xc42000e0f0)
	/usr/local/Cellar/go/1.7.3/libexec/src/runtime/panic.go:500 +0x1a1
github.com/coreos/kube-aws/config.(*EncryptedTLSAssets).Compact(0x0, 0xb, 0xc420126630, 0x9)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/tls_config.go:375 +0xc8
github.com/coreos/kube-aws/config.ReadOrCreateCompactTLSAssets(0xafcaf0, 0xb, 0xc420126630, 0x9, 0x0, 0x0, 0xc420013040, 0x4b, 0xc420126570, 0xa, ...)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/tls_config.go:437 +0x6a
github.com/coreos/kube-aws/config.Cluster.stackConfig(0xc420158240, 0x24, 0xafb398, 0x9, 0x1, 0x0, 0x0, 0x0, 0xc420126570, 0xa, ...)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/config.go:626 +0x186
github.com/coreos/kube-aws/config.Cluster.ValidateUserData(0xc420158240, 0x24, 0xafb398, 0x9, 0x1, 0x0, 0x0, 0x0, 0xc420126570, 0xa, ...)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/config.go:644 +0x77
github.com/coreos/kube-aws/cmd.runCmdValidate(0xee96a0, 0xf0acc0, 0x0, 0x0, 0x0, 0x0)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/cmd/validate.go:50 +0x19d
github.com/coreos/kube-aws/vendor/github.com/spf13/cobra.(*Command).execute(0xee96a0, 0xf0acc0, 0x0, 0x0, 0xee96a0, 0xf0acc0)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/vendor/github.com/spf13/cobra/command.go:632 +0x23e
github.com/coreos/kube-aws/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xee8e20, 0x0, 0x0, 0x0)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/vendor/github.com/spf13/cobra/command.go:722 +0x367
github.com/coreos/kube-aws/vendor/github.com/spf13/cobra.(*Command).Execute(0xee8e20, 0xc42014bf50, 0xc4200001a0)
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/vendor/github.com/spf13/cobra/command.go:681 +0x2b
main.main()
	/Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/main.go:9 +0x2d
~~~~

In the version from git (2b29fc34b00de630c627db6aadcd9eb42be66d0a) the message looks a bit different:
~~~~
$ $GOPATH/src/github.com/coreos/kube-aws/debug validate
Validating UserData...
Error: failed to render worker cloud config: template: .../userdata/cloud-config-worker:421:25: executing "userdata/cloud-config-worker" at <.TLSConfig.EtcdClien...>: can't evaluate field EtcdClientCert in type *config.CompactTLSAssets
~~~~

After some debugging it turns out that this message happens when the assets have not yet been encrypted, and there are errors in accessing KMS: ReadOrCreateEncryptedTLSAssets creates the KMS client, and then calls rawAssets.Encrypt. Encrypt doesn't actually check for errors properly, and instead just produces some form of emptiness (can't be more specific, this was my first attempt at reading and debugging go).

**Comment by mumoshu** [comment]
@ankon Thanks for your detailed report 👍 

This issue does seem to have existed and fixed for main clusters in #204 thanks to @redbaron now.

**Comment by flah00** [comment]
Also present for `kube-aws up --export` for 0.9.3-rc.2, when using my own CA.

```
AWS_PROFILE=zoo kube-aws up --export
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x96bc8]

goroutine 1 [running]:
panic(0x618520, 0xc420010150)
        /usr/local/Cellar/go/1.7.3/libexec/src/runtime/panic.go:500 +0x1a1
github.com/coreos/kube-aws/config.(*EncryptedTLSAssets).Compact(0x0, 0xb, 0xc42013c6b0, 0x9)
        /Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/tls_config.go:375 +0xc8
github.com/coreos/kube-aws/config.ReadOrCreateCompactTLSAssets(0x6ff046, 0xb, 0xc42013c6b0, 0x9, 0x0, 0x0, 0xc4200a4fa0, 0x4b, 0xc42013c5e8, 0x3, ...)
        /Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/tls_config.go:437 +0x6a
github.com/coreos/kube-aws/config.Cluster.stackConfig(0xc420170180, 0x21, 0xc42013c8d0, 0xa, 0x0, 0x0, 0x0, 0x0, 0xc42013c5e8, 0x3, ...)
        /Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/config.go:626 +0x186
github.com/coreos/kube-aws/config.Cluster.ValidateUserData(0xc420170180, 0x21, 0xc42013c8d0, 0xa, 0x0, 0x0, 0x0, 0x0, 0xc42013c5e8, 0x3, ...)
        /Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/config/config.go:644 +0x77
github.com/coreos/kube-aws/cmd.runCmdUp(0xaed5a0, 0xc42013c580, 0x0, 0x1, 0x0, 0x0)
        /Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/cmd/up.go:41 +0xb4
github.com/coreos/kube-aws/vendor/github.com/spf13/cobra.(*Command).execute(0xaed5a0, 0xc42013c560, 0x1, 0x1, 0xaed5a0, 0xc42013c560)
        /Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/vendor/github.com/spf13/cobra/command.go:632 +0x23e
github.com/coreos/kube-aws/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xaed160, 0x0, 0x0, 0x0)
        /Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/vendor/github.com/spf13/cobra/command.go:722 +0x367
github.com/coreos/kube-aws/vendor/github.com/spf13/cobra.(*Command).Execute(0xaed160, 0xc42015df50, 0xc4200001a0)
        /Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/vendor/github.com/spf13/cobra/command.go:681 +0x2b
main.main()
        /Users/cw_kuoka/Gopath/src/github.com/coreos/kube-aws/main.go:9 +0x2d
```

**Comment by mumoshu** [comment]
Closing this as the original issue is fixed since `v0.9.3-rc.3`. Please feel free to reopen the problem still persists.
